src/world.rs

```diff
@@ -121,6 +121,67 @@ impl World {
     }
 }
 
+impl World {
+    /// TODO
+    pub fn move_into(&mut self, ent: Entity, other: &mut World) -> Entity {
```

Comment:
I am rather unhappy about the amount of code duplication here. On the other hand, adjusting `alloc`, `free` and `move_` to be able to work in this context makes them significantly more complicated. Would probably need to factor out smaller reusable parts from these methods...